Packs/qualys/Integrations/Qualysv2/Qualysv2.py

```diff
@@ -2803,6 +2807,30 @@ def add_fields_to_events(events, time_field_path, event_type_field):
             event['event_type'] = event_type_field
 
 
+def truncate_asset_size(asset):
+    host_id = asset.get('ID') or 'NO_ID'
+    detection_id = asset.get('DETECTION', {}).get('UNIQUE_VULN_ID', 'No detection')
+
+    asset_size = get_size_of_object(asset)
+    if asset_size > ASSET_SIZE_LIMIT:
+        demisto.debug(f'{asset_size=}>{ASSET_SIZE_LIMIT=}')
+        detection_str = f' detection ID: {detection_id}' if detection_id else ''
+        demisto.debug(f'Asset ID: {host_id}{detection_str} has size of {asset_size}.')
+        results_characters_lim = 10000
+
+        if results := asset.get('DETECTION', {}).get('RESULTS'):
+            asset['DETECTION']['RESULTS'] = results[:results_characters_lim]
+            asset['isTruncated'] = True
+            demisto.debug(f'Truncated Asset ID: {host_id}{detection_str} to {results_characters_lim}')
+            demisto.debug(json.dumps(asset))
+
+        # For debug
+        for key, val in asset.items():
+            if (val_size := get_size_of_object(val)) > ASSET_SIZE_LIMIT:  # 1 MB
+                demisto.debug(f'Data under key "{key}" has size of {val_size}:\n'
+                              f'{str(val)[:10000]}...')
```

Comment:
We still need it?

Reply:
In theory, the fix should do the trick but I think that for future cases where other keys might be problematic, it could be nice to have a better overview of other problematic keys (probably will not happen but for any case that may come)